src/main/java/htsjdk/samtools/util/OverlapDetector.java

```diff
@@ -100,6 +103,7 @@ public void addLhs(final T object, final Locatable interval) {
      * and the corresponding objects.
      */
     public void addAll(final List<T> objects, final List<? extends Locatable> intervals) {
+        invalidateAllObjectsCache();
```

Comment:
This is unnecessary as this method calls `addLhs()` which also invalidates the cache.